frontend/src/app/page.tsx

```
@@ -48,6 +64,117 @@
+              onChange={(e) => setShadowRecovery(e.target.value)}
+            />
+          </label>
+        </div>
+
+        <button onClick={handleDownload}>Download HDR Image</button>
+
+        <div className="progress-bar">
+          <div
+            className="progress-bar-fill"
+            style={{ width: `${progress}%` }}
+          ></div>
```

Biome

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)